packages/m2m-gateway/src/services/eserviceTemplateService.ts

```diff
@@ -529,5 +529,105 @@ export function eserviceTemplateServiceBuilder(
 
       await pollEServiceTemplate(response, headers);
     },
+    async suspendEServiceTemplateVersion(
+      templateId: EServiceTemplateId,
+      versionId: EServiceTemplateVersionId,
+      { logger, headers }: WithLogger<M2MGatewayAppContext>
+    ): Promise<m2mGatewayApi.EServiceTemplateVersion> {
+      logger.info(
+        `Suspending version ${versionId} for eservice template with id ${templateId}`
+      );
+
+      const response =
+        await clients.eserviceTemplateProcessClient.suspendTemplateVersion(
+          undefined,
+          {
+            params: { templateId, templateVersionId: versionId },
+            headers,
+          }
+        );
+
+      const polledTemplate = await pollEServiceTemplate(
+        {
+          ...(await retrieveEServiceTemplateById(headers, templateId)),
+          metadata: response.metadata,
+        },
+        headers
+      );
+      const version = polledTemplate.data.versions.find(
+        (v) => v.id === versionId
+      );
+      if (!version) {
+        throw eserviceTemplateVersionNotFound(templateId, versionId);
+      }
+      return toM2MGatewayEServiceTemplateVersion(version);
+    },
+    async unsuspendEServiceTemplateVersion(
+      templateId: EServiceTemplateId,
+      versionId: EServiceTemplateVersionId,
+      { logger, headers }: WithLogger<M2MGatewayAppContext>
+    ): Promise<m2mGatewayApi.EServiceTemplateVersion> {
+      logger.info(
+        `Unsuspending version ${versionId} for eservice template with id ${templateId}`
+      );
+
+      const response =
+        await clients.eserviceTemplateProcessClient.activateTemplateVersion(
+          undefined,
+          {
+            params: { templateId, templateVersionId: versionId },
+            headers,
+          }
+        );
+
+      const polledTemplate = await pollEServiceTemplate(
+        {
+          ...(await retrieveEServiceTemplateById(headers, templateId)),
+          metadata: response.metadata,
+        },
+        headers
+      );
+      const version = polledTemplate.data.versions.find(
+        (v) => v.id === versionId
+      );
+      if (!version) {
+        throw eserviceTemplateVersionNotFound(templateId, versionId);
+      }
```

Comment:
Let's define a function that does that outside this function. Take [the eservice's one](https://github.com/pagopa/interop-be-monorepo/blob/9ce2b7cfa9fbd0106faa4e56595b6ff0b94d1351/packages/m2m-gateway/src/services/eserviceService.ts#L56) for example